src/services/stream.js

```diff
@@ -3,72 +3,85 @@ import stream from 'getstream'
+    let userToken = null
+    const get = () => userToken
+    const set = token => { userToken = token }
+    return Object.freeze({ get, set })
+  })()
+
+  // Feeds clients encapsulated
+  const feeds = (function () {
+    const data = {
+      achievement_aggregated: {
+        common: client.feed('achievement_aggregated', 'common', ACHIEVEMENT_COMMON_TOKEN)
+      },
+      timeline: {}
+    }
 
-  async function suscribeWithCallBacks (feedName, successCallback) {
+    const get = (feedName, group) => {
+      if (data[feedName][group]) {
+        return data[feedName][group]
+      }
+      const token = userToken.get()
+      if (!token) {
+        throw new Error('userToken is not available')
+      }
+      const feed = client.feed(feedName, group, token)
+      data[feedName][group] = feed
+      return feed
+    }
+
+    return Object.freeze({ get })
+  })()
+
+  async function suscribeWithCallBacks (feedName, group, successCallback) {
     try {
-      await feeds[feedName].subscribe(successCallback)
-      console.log(`Suscribed to getstream feed: ${feedName}`)
+      await feeds.get(feedName, group).subscribe(successCallback)
+      console.log(`Suscribed to getstream feed ${feedName}:${group}`)
     } catch (error) {
       console.log(error)
     }
   }
 
-  // For now, all data is fetched. Later, when filters will be implemented
-  // idea will be to fetch only the LIMIT first results of search, and have a system of pagination/scroll load
-  // while loop is temporary while data amount is low, and filters/search on their way to be implemented
-  // Imperative code below is thus temporary
-  async function fetchData (feedName, page) {
-    if (feedName === 'transactions') {
-      const { results, next } = await feeds[feedName].get({
-        limit: TRANSACTIONS_LIMIT,
-        offset: TRANSACTIONS_LIMIT * (page - 1)
-      })
-      return { results, hasMore: next !== '' }
-    }
+  async function fetchData (feedName, group, page = 1) {
+    const { results, next } = await feeds.get(feedName, group).get({
+      limit: LIMIT,
+      offset: LIMIT * (page - 1)
+    })
+    console.log(`Fetch succeeded for ${feedName}:${group}`, results)
+    return { results, hasMore: next !== '' }
+  }
+
+  async function setUser ({ data, userAddress }) {
+    const client = userClient.get()
     try {
-      let items
-      const { next, results } = await feeds[feedName].get({ limit: LIMIT })
-      items = results
-      let allFetched = next === ''
-      if (!allFetched) {
-        let previousRequestsCount = 1
-        while (!allFetched) {
-          const { results: newResults, next: newNext } = await feeds[feedName].get({
-            limit: LIMIT,
-            offset: LIMIT * previousRequestsCount
-          })
-          previousRequestsCount += 1
-          allFetched = newNext === ''
-          items = [ ...items, ...newResults ]
-        }
-      }
-      return { results: items }
+      await client.user(userAddress).update(data)
```

Comment:
So when user authorize with another social profile, his wallet will be associated with the new profile, in all places will be shown his new name?

Reply:
Yes